plugins/ngrok/provisioner.go

```diff
@@ -24,7 +27,25 @@ const (
 type fileProvisioner struct {
 }
 
-func newNgrokProvisioner() sdk.Provisioner {
+func ngrokProvisioner() sdk.Provisioner {
+	currentVersion, requiredVersion, err := getNgrokVersion()
+	if err != nil {
+		// When ngrok version check fails for any reason,
+		// use config file to provision as a fallback
+		return fileProvisioner{}
+	}
+
+	// If the current ngrok CLI version is 3.2.1 or higher,
+	// use environment variables to provision credentials
+	//
+	// semver.Compare resulting in 0 means 3.2.1 is in use
+	// semver.Compare resulting in +1 means >3.2.1 is in use
+	if semver.Compare(currentVersion, requiredVersion) == 0 || semver.Compare(currentVersion, requiredVersion) == +1 {
+		newNgrokEnvVarProvisioner := ngrokEnvVarProvisioner{}
+		newNgrokEnvVarProvisioner.Provision(context.Background(), sdk.ProvisionInput{}, &sdk.ProvisionOutput{})
```

Comment:
At this point we should not call the `Provision` method but rather just return the `ngrokEnvVarProvisioner{}`. 

Provisioning is generally never done inside the shell plugin code. The CLI code must call the `Provision` function, and the shell plugin's code reponsibility is to provide the CLI with the right provisioner to call.